services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -7824,6 +7826,38 @@ public Optional<PushStatusStoreWriter> getPushStatusStoreWriter() {
     return pushStatusStoreWriter;
   }
 
+  @Override
+  public void sendHeartbeatToSystemStore(String clusterName, String storeName, long heartbeatTimeStamp) {
+    VeniceSystemStoreType systemStoreType = VeniceSystemStoreType.getSystemStoreType(storeName);
```

Comment:
`VeniceSystemStoreType.getSystemStoreType` can return `BATCH_JOB_HEARTBEAT_STORE` or `null` if there is no match.
To guard these cases, we should do a `null` check, and only send heartbeat for `DAVINCI_PUSH_STATUS_STORE` and `META_STORE`, and for `  BATCH_JOB_HEARTBEAT_STORE`, this function should fail fast.

This comment applies to function: `getHeartbeatFromSystemStore` as well.

Reply:
I think instead of checking them here, let me add another safety check in the 
checkAndSendHeartbeatToSystemStores() which has iterations over stores. 

I refactored that method a bit to contain the check. So it will first check if it is a system store, and then check if it is a user system store.